genesyscloud/routing_queue_outbound_email_address/genesyscloud_routing_queue_outbound_email_address_proxy.go

```diff
@@ -0,0 +1,165 @@
+
+// routingQueueOutboundEmailAddressProxy contains all of the methods that call genesys cloud APIs.
+type routingQueueOutboundEmailAddressProxy struct {
+	clientConfig                               *platformclientv2.Configuration
+	routingApi                                 *platformclientv2.RoutingApi
+	getAllRoutingQueuesAttr                    getAllRoutingQueuesFunc
+	getRoutingQueueOutboundEmailAddressAttr    getRoutingQueueOutboundEmailAddressFunc
+	updateRoutingQueueOutboundEmailAddressAttr updateRoutingQueueOutboundEmailAddressFunc
+}
+
+// newRoutingQueueOutboundEmailAddressProxy initializes the Routing queue outbound email address proxy with the data needed to communicate with Genesys Cloud
+func newRoutingQueueOutboundEmailAddressProxy(clientConfig *platformclientv2.Configuration) *routingQueueOutboundEmailAddressProxy {
+	api := platformclientv2.NewRoutingApiWithConfig(clientConfig)
+	return &routingQueueOutboundEmailAddressProxy{
+		clientConfig:                               clientConfig,
+		routingApi:                                 api,
+		getAllRoutingQueuesAttr:                    getAllRoutingQueuesFn,
+		getRoutingQueueOutboundEmailAddressAttr:    getRoutingQueueOutboundEmailAddressFn,
+		updateRoutingQueueOutboundEmailAddressAttr: updateRoutingQueueOutboundEmailAddressFn,
+	}
+}
+
+func getRoutingQueueOutboundEmailAddressProxy(clientConfig *platformclientv2.Configuration) *routingQueueOutboundEmailAddressProxy {
+	if internalProxy == nil {
+		internalProxy = newRoutingQueueOutboundEmailAddressProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getAllRoutingQueues gets all routing queues in an org
+func (p *routingQueueOutboundEmailAddressProxy) getAllRoutingQueues(ctx context.Context) (*[]platformclientv2.Queue, *platformclientv2.APIResponse, error) {
+	return p.getAllRoutingQueuesAttr(ctx, p)
+}
+
+// getRoutingQueueOutboundEmailAddress gets the Outbound Email Address for a queue
+func (p *routingQueueOutboundEmailAddressProxy) getRoutingQueueOutboundEmailAddress(ctx context.Context, queueId string) (*platformclientv2.Queueemailaddress, *platformclientv2.APIResponse, error) {
+	return p.getRoutingQueueOutboundEmailAddressAttr(ctx, p, queueId)
+}
+
+// updateRoutingQueueOutboundEmailAddress updates the Outbound Email Address for a queue
+func (p *routingQueueOutboundEmailAddressProxy) updateRoutingQueueOutboundEmailAddress(ctx context.Context, queueId string, rules *platformclientv2.Queueemailaddress) (*platformclientv2.Queueemailaddress, *platformclientv2.APIResponse, error) {
+	return p.updateRoutingQueueOutboundEmailAddressAttr(ctx, p, queueId, rules)
+}
+
+// getAllRoutingQueuesFn is an implementation function for getting all queues in an org
+func getAllRoutingQueuesFn(ctx context.Context, p *routingQueueOutboundEmailAddressProxy) (*[]platformclientv2.Queue, *platformclientv2.APIResponse, error) {
```

Comment:
@dginty4 Can we add caching to this git all code so we can get this out of the wya

Reply:
Just added caching